src/main/kotlin/org/digma/intellij/plugin/ui/common/EnvironmentsPanel.kt

```diff
@@ -170,27 +170,26 @@ class EnvironmentsPanel(
     }
 
     private fun rebuild() {
-
-        if (components.isNotEmpty()) {
-            this.components.forEach {
-                this.remove(it)
+        SwingUtilities.invokeLater {
+            if (components.isNotEmpty()) {
+                this.components.forEach {
+                    this.remove(it)
+                }
             }
-        }
 
-        val usageStatusResult = model.getUsageStatus()
+            val usageStatusResult = model.getUsageStatus()
 
-        val envsThatHaveUsageSet: Set<String> = buildEnvironmentWithUsages(usageStatusResult)
-        val hasUsageFunction = fun(env: String): Boolean { return envsThatHaveUsageSet.contains(env) }
+            val envsThatHaveUsageSet: Set<String> = buildEnvironmentWithUsages(usageStatusResult)
+            val hasUsageFunction = fun(env: String): Boolean { return envsThatHaveUsageSet.contains(env) }
 
-        val relevantEnvs = buildRelevantSortedEnvironments(environmentsSupplier, hasUsageFunction)
+            val relevantEnvs = buildRelevantSortedEnvironments(environmentsSupplier, hasUsageFunction)
 
 
-        for (currEnv in relevantEnvs) {
-            val isSelectedEnv = currEnv.contentEquals(environmentsSupplier.getCurrent())
-            val toolTip = buildToolTip(usageStatusResult, currEnv)
-            val linkText = buildLinkText(currEnv, isSelectedEnv)
+            for (currEnv in relevantEnvs) {
+                val isSelectedEnv = currEnv.contentEquals(environmentsSupplier.getCurrent())
+                val toolTip = buildToolTip(usageStatusResult, currEnv)
+                val linkText = buildLinkText(currEnv, isSelectedEnv)
 
-            SwingUtilities.invokeLater {
                 buildEnvironmentsPanelButtons(currEnv, linkText, isSelectedEnv, toolTip, hasUsageFunction)
```

Comment:
I think its better to remove the components here just before building the buttons

Reply:
got your point! thanks!